docs/cadl-for-openapi-dev.md

```diff
@@ -0,0 +1,285 @@
+# Cadl for the OpenAPI developer
+
+This guide is an introduction to Cadl using concepts that will be familiar to developers
+that either build or use API definitions in OpenAPI v2 or v3.
+
+In many case, this will also describe how the cadl-autorest and openapi3 emitters translate
+Cadl designs into OpenAPI.
+
+The document is organized around the features of an OpenAPI v2 or v3 definition.
+The idea is that if you know how to describe some API feature in OpenAPI, you can just navigate
+to the section of this document for that feature.
+
+## Data Types
+
+In OpenAPI [v2](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md#data-types)/[v3](https://github.com/OAI/OpenAPI-Specification/blob/3.0.3/versions/3.0.3.md#dataTypes), data types are specified using the `type` and `format` fields in a schema.
+
+The Cadl equivalent of OpenAPI data types are the Cadl primitive types or [built-in models](https://github.com/microsoft/cadl/blob/main/docs/tutorial.md#built-in-models).
+
+The following table shows how common OpenAPI types map to Cadl types:
+
+| OpenAPI `type`/`format`           | Cadl type        | Notes                  |
+| --------------------------------- | ---------------- | ---------------------- |
+| `type: integer, format: int32`    | `int32`          |                        |
+| `type: integer, format: int64`    | `int64`          |                        |
+| `type: number, format: float`     | `float32`        |                        |
+| `type: number, format: double`    | `float64`        |                        |
+| `type: string`                    | `string`         |                        |
+| `type: string, format: byte`      |                  |                        |
+| `type: string, format: binary`    | `byte[]`         | any sequence of octets |
```

Comment:
The ```byte[]``` type should correspond to ```type: string, format: byte```, it is meant to be interpreted as a base64-encoded string.  We don't yet have a representation for a stream of octets.

Reply:
Also, I think I replaced byte with bytes some time ago?